src/plugins/dataflow-tool/components/ui/dataflow-program-topbar.tsx

```diff
@@ -145,22 +145,30 @@ export const DataflowProgramTopbar = (props: TopbarProps) => {
         </div>
       </div>
       <div className="topbar-center">
+
         <RateSelectorComponent
           rateOptions={props.programDataRates}
           dataRate={props.dataRate}
           onRateSelectClick={props.onRateSelectClick}
           readOnly={props.readOnly}
         />
-        <RecordButton readOnly={props.readOnly} />
-        <button
-          className="program-state-button"
-          title="Stop Program"
-          onClick={props.onStopProgramClick}
-          disabled={!props.runningProgram || !props.readOnly}
-        >
-          <div className="icon stop" />
-          <div className="text">Stop</div>
-        </button>
+
+        { props.showRecordUI &&
+          <RecordButton readOnly={props.readOnly} />
+        }
+
+        { props.showRecordUI &&
```

Comment:
If you wanted to have just one check on `props.showRecordUI`, you could wrap the `RecordButton` and `button` in `<> ... </>`.